FeedStoreChallenge/ManagedCache.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  ManagedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Darren Findlay on 29/05/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+@objc(ManagedCache)
+class ManagedCache: NSManagedObject {
+	@NSManaged var timestamp: Date
+	@NSManaged var feed: NSOrderedSet
+
+	internal static func newSingleCache(in context: NSManagedObjectContext) throws -> ManagedCache {
+		try findFirst(in: context).map(context.delete)
+		return ManagedCache(context: context)
+	}
+
+	static func findFirst(in context: NSManagedObjectContext) throws -> ManagedCache? {
+		let request = NSFetchRequest<ManagedCache>(entityName: ManagedCache.entity().name!)
+		return try context.fetch(request).first
+	}
+
+	var localFeed: [LocalFeedImage] {
+		feed.map { $0 as! ManagedFeedImage }
```

Comment:
Instead of `map` with force cast, it could be `compactMap` with optional casting

```suggestion
		feed.compactMap { $0 as? ManagedFeedImage }
```